dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -191,8 +191,14 @@ protected void run(@Nonnull Connector connector, @Nonnull ConnectorArguments arg
         // We had a customer request to base it on the database, but that isn't well-defined,
         // as there may be 0 or N databases in a single file.
         File outputFile = arguments.getOutputFile();
+        String outputDirectory = arguments.getOutputDirectory();
+
+        if ((outputDirectory == null) == (outputFile == null)) {
+            throw new IllegalStateException("Please use exactly one of --output & --output-dir flags");
```

Comment:
I'm sorry, I wasn't very precise in my previous explanation "only one of the two should be allowed". What I missed to mention is that we also still want to accept no options.

We need to handle 3 possibilities:
1. Not `--output` nor `--output-dir` is used. This should be the default behavior as it worked before and it should not change. The file should be dumped in the current directory with the default filename.
2. Only `--output` is used. This should work the same as it worked before, i.e. if the argument is a filename ending with `.zip`, then dump to this file, otherwise dump to the directory or return an error. No change in the logic in this case.
3. Only `--output-dir` is used. This is the new logic being introduced in this PR. A default zip filename should be appended to the argument of this option and the result should be dumped to this zip file (i.e. a default zip file will be created in the directory provided by the `--output-dir` option).

If both options are used by mistake, then display an error and return immediately. Instead of throwing an exception, please, display an error in the user-friendly way (with the stars around), similarly to:
https://github.com/google/dwh-migration-tools/blob/ed75f929f475b27701bd71c3da120292f10b1127/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java#L308